### ISSUE TITLE
[Wildcard Variables] support for `avoid_types_as_parameter_names`

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#5051]
This is wrongly reporting:

```dart
class C<_> {
  var _;
  C.c(this._, _);
}
```

**Comment by pq** [comment]
Fixed w/ https://github.com/dart-lang/sdk/commit/a9c63ea899d236bebbee6dabac1b5f08b0918c41
